subprojects/packagefiles/speexdsp/meson.build

```diff
@@ -0,0 +1,85 @@
+ project('speexdsp', 'c',
+  version : '1.2.0',
+  default_options: ['warning_level=2',
+                    'c_std=gnu99',
+                    'buildtype=release'],
+  license : 'bsd-like')
+
+lib_current = 1
+lib_revision = 0
+lib_age = 2
+lib_soversion = 1
+lib_version = '@0@.@1@.@2@'.format(lib_soversion, lib_age, lib_revision)
+
+cc = meson.get_compiler('c')
+host_cpu_family = host_machine.cpu_family()
+
+add_project_arguments('-DHAVE_CONFIG_H', language: 'c')
+
+conf_data = configuration_data()
+
+# Check for SIMD instructions
+c_args = []
+if host_cpu_family.startswith('x86')
+  c_args += cc.get_supported_arguments(['-mfpmath=sse', '-msse2'])
+  conf_data.set10('USE_SSE', true)
+  conf_data.set10('USE_SSE2', true)
+
+elif host_cpu_family == 'arm'
+  c_args += cc.get_supported_arguments(['-mfpu=neon-vfpv4'])
+  conf_data.set10('USE_NEON', true)
+
+elif host_cpu_family == 'aarch64'
+  c_args += cc.get_supported_arguments(['-mfpu=neon-fp-armv8'])
+  conf_data.set10('USE_NEON', true)
+endif
+
+foreach arg : c_args
+  if cc.has_argument(arg)
+    add_project_arguments(arg, language: 'c')
+  endif
+endforeach
+
+# Check for C99 variable-size arrays, or alloca() as fallback
+msg_use_alloca = false
+if cc.compiles('''static int x;
+                  char some_func (void) {
+                    char a[++x];
+                    a[sizeof a - 1] = 0;
+                    int N;
+                    return a[0];
+                  }''', name : 'C99 variable-size arrays')
+  conf_data.set('VAR_ARRAYS', 1)
+  msg_use_alloca = 'NO (using C99 variable-size arrays instead)'
+elif cc.compiles('''#include <alloca.h>
+                    void some_func (void) {
+                      int foo=10;
+                      int * array = alloca(foo);
+                    }''', name : 'alloca (alloca.h)')
+  conf_data.set('USE_ALLOCA', true)
+  conf_data.set('HAVE_ALLOCA_H', true)
+  msg_use_alloca = true
+elif cc.compiles('''#include <malloc.h>
+                    #include <stdlib.h>
+                    void some_func (void) {
+                      int foo=10;
+                      int * array = alloca(foo);
+                    }''', name : 'alloca (std)')
+  conf_data.set('USE_ALLOCA', true)
+  msg_use_alloca = true
+endif
+
+# Symbol visibility definition
+if cc.get_id() == 'msvc'
+  conf_data.set('EXPORT', '')
```

Comment:
You can use the dllexport declspec here, surely?

Reply:
Yes; adding.

Reply:
Tried, along with a couple variations but the builds fails with

```text
error C2375: 'speex_preprocess_state_init': redefinition; different linkage
```

(For every function)

Checked how vcpkg is handling the `EXPORT` define, and they also make it an empty string for msvc:

https://github.com/microsoft/vcpkg/blob/314018fef43c7e3a52a3e46599313b70ace3977a/ports/speexdsp/CMakeLists.txt#L53

Going with that.

Reply:
Ah, I get it now.

On GCC / clang, they use visibility attributes, and on MSVC they just use the vs_module_defs file which I see you already included. Makes sense. I think upstream should not require `define`ing this though, sad...